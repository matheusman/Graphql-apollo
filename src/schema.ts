import { gql } from "apollo-server";
import { userTypeDefs } from "./graphql/user/userTypeDefs";
import { userResolvers } from "./graphql/user/userResolvers";
import { postTypeDefs } from "./graphql/posts/postTypeDefs";
import { postResolvers } from "./graphql/posts/postResolvers";

const rootTypeDefs = gql`
  type Query {
    _isEmpty: Boolean!
  }
`;

const rootResolvers = {
  Query: {
    _isEmpty: () => true,
  },
};

export const typeDefs = [rootTypeDefs, userTypeDefs, postTypeDefs]
export const resolvers = [rootResolvers, userResolvers, postResolvers]
