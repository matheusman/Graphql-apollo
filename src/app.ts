import { Server } from "./index";

export const app = new Server();

app.startServer();
