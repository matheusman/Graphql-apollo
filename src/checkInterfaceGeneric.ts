export function checkInterface<T>(id: unknown, ...props: [keyof T]): id is T {
  if (
    id &&
    typeof id === 'object' &&
    props.filter((values) => values in id).length >= props.length
  ) {
    return true;
  } else {
    return false;
  }
}
