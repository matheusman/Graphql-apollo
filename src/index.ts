import { ApolloServer } from 'apollo-server';
import { resolvers, typeDefs } from './schema';
import { context } from './graphql/context/context';
import 'dotenv/config';
import { dataSources } from './graphql/datasources/dataSources';

export const server = new ApolloServer({
	typeDefs,
	resolvers,
	context,
  dataSources: () => {
    return {
      ...dataSources
    }
  }
});

