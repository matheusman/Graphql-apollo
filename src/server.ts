import { server } from ".";

server.listen(4003, () => console.log('server running on port ' + 4003));
