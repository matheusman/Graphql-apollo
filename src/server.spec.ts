import { gql } from 'apollo-server';
import { server } from '../src';

describe('should server its be defined', () => {
	it('should execute query is empty', async () => {
		const testing = await server.executeOperation({
			query: `query isEmpty {
        _isEmpty
      }`,
		});
    expect(testing).toBeDefined()
    expect(testing.data?._isEmpty).toBeTruthy()
  });

  it('shoud server mutation is empty and defined', async () => {
    const testing = await server.executeOperation({
			query: `mutation isEmpty {
        _isEmpty
      }`,
		});
		expect(testing).toBeDefined();
		expect(testing.data?._isEmpty).toBeTruthy();
  })

  it('match snapshot', async () => {
    const query = await server.executeOperation({
			query: `query isEmpty {
        _isEmpty
      }`,
		});
    const mutation = await server.executeOperation({
			query: `mutation isEmpty {
        _isEmpty
      }`,
		});
    expect({ query: query, mutation: mutation}).toMatchSnapshot()
  })
});
