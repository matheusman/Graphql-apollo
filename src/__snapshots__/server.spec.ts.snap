// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should server its be defined match snapshot 1`] = `
{
  "mutation": {
    "data": {
      "_isEmpty": true,
    },
    "errors": undefined,
    "extensions": undefined,
    "http": {
      "headers": Headers {
        Symbol(map): {},
      },
    },
  },
  "query": {
    "data": {
      "_isEmpty": true,
    },
    "errors": undefined,
    "extensions": undefined,
    "http": {
      "headers": Headers {
        Symbol(map): {},
      },
    },
  },
}
`;
